go-controller/pkg/metrics/metrics.go

```diff
@@ -144,6 +172,177 @@ func coverageShowMetricsUpdater(component string) {
+			Namespace: metricNamespace,
+			Subsystem: metricSubsystem,
+			Name:      fmt.Sprintf("%s_short_term_avg", metricName),
+		})
+
+		metricInfo.metrics.longTermAvg = prometheus.NewGauge(prometheus.GaugeOpts{
+			Namespace: metricNamespace,
+			Subsystem: metricSubsystem,
+			Name:      fmt.Sprintf("%s_long_term_avg", metricName),
+		})
+
+		prometheus.MustRegister(metricInfo.metrics.totalSamples)
+		prometheus.MustRegister(metricInfo.metrics.min)
+		prometheus.MustRegister(metricInfo.metrics.max)
+		prometheus.MustRegister(metricInfo.metrics.percentile95th)
+		prometheus.MustRegister(metricInfo.metrics.shortTermAvg)
+		prometheus.MustRegister(metricInfo.metrics.longTermAvg)
+	}
+}
+
+// getStopwatchShowOutputMap obtains the stopwatch/show metric values for the specified component.
+func getStopwatchShowOutputMap(component string) (map[string]stopwatchStatistics, error) {
+	var stdout, stderr string
+	var err error
+
+	defer func() {
+		if r := recover(); r != nil {
+			err = fmt.Errorf("recovering from a panic while parsing the stopwatch/show output for %s: %v", component, r)
+		}
+	}()
+
+	switch component {
+	case ovnController:
+		stdout, stderr, err = util.RunOVNControllerAppCtl("stopwatch/show")
+	case ovnNorthd:
+		stdout, stderr, err = util.RunOVNNorthAppCtl("stopwatch/show")
+	default:
+		return nil, fmt.Errorf("unknown component %s for stopwatch/show", component)
+	}
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to get stopwatch/show output for %s (stderr: %s): %w", component, stderr, err)
+	}
+
+	parsedOutput := parseStopwatchShowOutput(stdout)
+	return parsedOutput, err //need to return err, as it could be set in the defer
+}
+
+// parseStopwatchShowOutput returns the number of total samples for each poll loop
+func parseStopwatchShowOutput(output string) map[string]stopwatchStatistics {
+	result := make(map[string]stopwatchStatistics)
+
+	reStatisticsBlock := regexp.MustCompile(`(?m)^Statistics for '(?P<name>.+)'$(?:\n  .*)+`)
+	reTotalSamples := regexp.MustCompile(`(?m)^  Total samples: (?P<totalSamples>\d*)$`)
+	reMaximum := regexp.MustCompile(`(?m)^  Maximum: (?P<max>\d*) msec$`)
+	reMinumum := regexp.MustCompile(`(?m)^  Minimum: (?P<min>\d*) msec$`)
+	rePercentile95th := regexp.MustCompile(`(?m)^  95th percentile: (?P<percentile95th>\d*\.?\d*) msec$`)
+	reShortTermAvg := regexp.MustCompile(`(?m)^  Short term average: (?P<shortTermAvg>\d*\.?\d*) msec$`)
+	reLongTermAvg := regexp.MustCompile(`(?m)^  Long term average: (?P<longTermAvg>\d*\.?\d*) msec$`)
+
+	nameIndex := reStatisticsBlock.SubexpIndex("name")
+	totalSamplesIndex := reTotalSamples.SubexpIndex("totalSamples")
+	maxIndex := reMaximum.SubexpIndex("max")
+	minIndex := reMinumum.SubexpIndex("min")
+	percentile95thIndex := rePercentile95th.SubexpIndex("percentile95th")
+	shortTermAvgIndex := reShortTermAvg.SubexpIndex("shortTermAvg")
+	longTermAvgIndex := reLongTermAvg.SubexpIndex("longTermAvg")
```

Comment:
This doesn't need to be done during each iteration, right? This can be moved to the top of the file in a `var` clause.